### ISSUE TITLE
Keybinding error

### ISSUE THREAD

**Issue by 40P3** [brittyazel/Neuron#189]
Hi,

Noticed on my hunter that when I pushed my mount button it used a different mount than it was on neuron's macro. Seems like the game's main bar keybinding is active in the background (thus a normal blizzard macro mount was the button using). Then I reallized I got no button keybinding setup in neuron. When I tried to set buttons I saw this error:

24x Neuron\Neuron-KeyBinder.lua:235: attempt to index field 'hotKeys' (a nil value)
Neuron\Neuron-KeyBinder.lua:235: in function `ProcessBinding'
Neuron\Neuron-KeyBinder.lua:397: in function `OnKeyDown'
Neuron\Neuron-KeyBinder.lua:434: in function <Neuron\Neuron-KeyBinder.lua:434>

Locals:
self = <table> {
 SetDefaultModuleLibraries = <function> defined @Accountant\Libs\AceAddon-3.0\AceAddon-3.0.lua:398
 Enable = <function> defined @Accountant\Libs\AceAddon-3.0\AceAddon-3.0.lua:325
 NewModule = <function> defined @Accountant\Libs\AceAddon-3.0\AceAddon-3.0.lua:266
 EnableModule = <function> defined @Accountant\Libs\AceAddon-3.0\AceAddon-3.0.lua:363
 modules = <table> {
 }
 GetModule = <function> defined @Accountant\Libs\AceAddon-3.0\AceAddon-3.0.lua:241
 IterateEmbeds = <function> defined @Accountant\Libs\AceAddon-3.0\AceAddon-3.0.lua:473
 SecureHook = <function> defined @ArkInventory\Externals\Ace3\AceHook\AceHook-3.0.lua:336
 defaultModuleLibraries = <table> {
 }
 OnUpdate = <function> defined @Neuron\Neuron-KeyBinder.lua:326
 baseName = "Neuron"
 CreateBindFrame = <function> defined @Neuron\Neuron-KeyBinder.lua:421
 RawHookScript = <function> defined @ArkInventory\Externals\Ace3\AceHook\AceHook-3.0.lua:395
 OnShow = <function> defined @Neuron\Neuron-KeyBinder.lua:250
 OnMouseWheel = <function> defined @Neuron\Neuron-KeyBinder.lua:403
 OnKeyDown = <function> defined @Neuron\Neuron-KeyBinder.lua:386
 OnClick = <function> defined @Neuron\Neuron-KeyBinder.lua:339
 SetDefaultModulePrototype = <function> defined @Accountant\Libs\AceAddon-3.0\AceAddon-3.0.lua:440
 OnLeave = <function> defined @Neuron\Neuron-KeyBinder.lua:319
 ClearBindings = <function> defined @Neuron\Neuron-KeyBinder.lua:144
 OnEnter = <function> defined @Neuron\Neuron-KeyBinder.lua:280
 name = "Neuron_Binder"
 ApplyBindings = <function> defined @Neuron\Neuron-KeyBinder.lua:190
 IsEnabled = <function> defined @Accountant\Libs\AceAddon-3.0\AceAddon-3.0.lua:482
 orderedModules = <table> {
 }
 OnHide = <function> defined @Neuron\Neuron-KeyBinder.lua:275
 DisableModule = <function> defined @Accountant\Libs\AceAddon-3.0\AceAddon-3.0.lua:381
 ProcessBinding = <function> defined @Neuron\Neuron-KeyBinder.lua:224
 GetKeyText = <function> defined @Neuron\Neuron-KeyBinder.lua:105
 SetNeuronBinding = <function> defined @Neuron\Neuron-KeyBinder.lua:172
 GetName = <function> defined @Accountant\Libs\AceAddon-3.0\AceAddon-3.0.lua:310
 RegisterMessage = <function> defined @Accountant\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:116
 UnregisterMessage = <function> defined @Accountant\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:179
 GetModifier = <function> defined @Neuron\Neuron-KeyBinder.lua:55
 OnDisable = <function> defined @Neuron\Neuron-KeyBinder.lua:44
 IsHooked = <function> defined @ArkInventory\Externals\Ace3\AceHook\AceHook-3.0.lua:493
 OnEnable = <function> defined @Neuron\Neuron-KeyBinder.lua:35
 OnInitialize = <function> defined @Neuron\Neuron-KeyBinder.lua:26
 IsModule = <function> defined @Accountant\Libs\AceAddon-3.0\AceAddon-3.0.lua:248
 hooks = <table> {
 }
 SetDefaultModuleState = <function> defined @Accountant\Libs\AceAddon-3.0\AceAddon-3.0.lua:418
 SendMessage = <function> defined @Accountant\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:85
 UnhookAll = <function> defined @ArkInventory\Externals\Ace3\AceHook\AceHook-3.0.lua:477
 SetEnabledState = <function> defined @Accountant\Libs\AceAddon-3.0\AceAddon-3.0.lua:455
 Hook = <function> defined @ArkInventory\Externals\Ace3\AceHook\AceHook-3.0.lua:274
 enabledState = true
 Unhook = <function> defined @ArkInventory\Externals\Ace3\AceHook\AceHook-3.0.lua:418
 RegisterEvent = <function> defined @Accou

To be fair on this tune I have a different UI setup than on most of my toon, I set up the Overwatch UI OwD. Maybe its a conflict, but on other char (yesterday played my paladin) I didn't notice this error yet.

**Comment by 40P3** [comment]
To be fair, after /reloadui I was able to set the keybinding on Neuron.

**Comment by brittyazel** [comment]
Yes the /reload fix was in the update message, but there shouldn't have
been an error. Are you running the most up to date version? This bug looks
like the same one reported a few days ago that has since been fixed

**Comment by 40P3** [comment]
Last wednesday update, 0.9.37a

**Comment by brittyazel** [comment]
Is this still an issue?

**Comment by 40P3** [comment]
Hi,

Thanks for getting back to me about this. For some reason I had to set up all my chars like above once, then the keybinding stayed and no errors.
If I would make a new char and enable neuron I believe the error would occur once, until I set up keybinding and do a Reload.
Can you try it? I will try it once I get a little time today.
